src/Payments.sol

```diff
@@ -0,0 +1,612 @@
+
+    modifier validateRailActive(uint256 railId) {
+        require(rails[railId].from != address(0), "rail does not exist");
+        require(rails[railId].isActive, "rail is inactive");
+        _;
+    }
+
+    modifier validateRailAccountsExist(uint256 railId) {
+        Rail storage rail = rails[railId];
+        require(rail.from != address(0), "rail does not exist");
+        require(accounts[rail.token][rail.from].ownerAddress != address(0), "from account does not exist");
+        require(accounts[rail.token][rail.to].ownerAddress != address(0), "to account does not exist");
+        _;
+    }
+
+    modifier onlyRailOperator(uint256 railId) {
+        require(rails[railId].operator == msg.sender, "only the rail operator can perform this action");
+        _;
+    }
+
+    modifier onlyAccountOwner(address token) {
+        address owner = accounts[token][msg.sender].ownerAddress;
+        require(owner != address(0), "account does not exist");
+        require(owner == msg.sender, "not account owner");
+        _;
+    }
+
+    function approveOperator(
+        address token,
+        address operator,
+        address arbiter,
+        uint256 rateAllowance,
+        uint256 lockupAllowance
+    ) external onlyAccountOwner(token) {
+        require(token != address(0), "token address cannot be zero");
+        require(operator != address(0), "operator address cannot be zero");
+
+        OperatorApproval storage approval = operatorApprovals[token][msg.sender][operator];
+        approval.arbitrer = arbiter;
+        approval.rateAllowance = rateAllowance;
+        approval.lockupAllowance = lockupAllowance;
+        approval.isApproved = true;
+    }
+
+    // TODO: Revisit
+    function terminateOperator(address operator) external  {
+        require(operator != address(0), "operator address invalid");
+
+        uint256[] memory railIds = clientOperatorRails[msg.sender][operator];
+        for (uint256 i = 0; i < railIds.length; i++) {
+            Rail storage rail = rails[railIds[i]];
+            require(rail.from == msg.sender, "Not rail payer");
+            if (!rail.isActive) {
+                continue;
+            }
+
+            settleRail(railIds[i], block.number);
+
+            Account storage account = accounts[rail.token][msg.sender];
+            account.lockupCurrent -= rail.lockupFixed + (rail.paymentRate * rail.lockupPeriod);
+            account.lockupRate -= rail.paymentRate;
+
+            rail.paymentRate = 0;
+            rail.lockupFixed = 0;
+            rail.lockupPeriod = 0;
+            rail.isActive = false;
+
+            OperatorApproval storage approval = operatorApprovals[rail.token][msg.sender][operator];
+            approval.rateAllowance = 0;
+            approval.lockupAllowance = 0;
+            approval.isApproved = false;
+        }
+    }
+
+    // TODO: implement
+    function terminateRail() public {
```

Comment:
An operator can always do this by setting everything to 0.

Reply:
If a client wants to terminate a rail -> they go the operator and get them to modify the rates and lockups for the rail to 0.

Reply:
Removed this function.